app/server.go

```diff
@@ -1204,6 +1189,14 @@ func stripPort(hostport string) string {
 }
 
 func (s *Server) Start() error {
+	// Start products.
+	// This needs to happen before because products are dependent on main server.
```

Comment:
This sentence doesn't quite parse to me: if products are dependent on the server, how can they start before?

Reply:
Now that I read it, it does sound odd :stuck_out_tongue: 

Maybe "dependent" is not the right word here in the context of "starting". Products do take the `*Server` as a dependency in their constructor. But while starting, we need to start the products first, and then start the main HTTP server. Because otherwise, the HTTP server will be ready to respond to requests, but products won't be, since they haven't started.

I am not sure if there's a better one-liner to explain this?

Reply:
Maybe we can just call it http server instead of main server?

Reply:
I like that :+1: